native/src/main/java/io/ballerina/stdlib/data/csvdata/utils/CsvUtils.java

```diff
@@ -0,0 +1,350 @@
+                        String originalName = key.substring(Constants.FIELD.length());
+                        if (updatedValues.contains(name)) {
+                            throw DiagnosticLog.error(DiagnosticErrorCode.DUPLICATE_FIELD, name);
+                        }
+                        if (updatedFields.contains(originalName)) {
+                            throw DiagnosticLog.error(DiagnosticErrorCode.DUPLICATE_FIELD, originalName);
+                        }
+                        updatedFields.add(originalName);
+                        updatedValues.add(name);
+                        updatedRecordFieldNames.put(name, originalName);
+                        break;
+                    }
+                }
+            }
+        }
+        for (String field : fieldHierarchy.keySet()) {
+            if (updatedFields.contains(field)) {
+                continue;
+            }
+            if (updatedValues.contains(field) || updatedRecordFieldNames.containsKey(field)) {
+                throw DiagnosticLog.error(DiagnosticErrorCode.DUPLICATE_FIELD, field);
+            }
+            updatedRecordFieldNames.put(field, field);
+        }
+
+        return updatedRecordFieldNames;
+    }
+
+    public static String getUpdatedHeaders(Map<String, String> updatedRecords, String key, boolean isKeyContains) {
+        String fieldName = updatedRecords.get(key);
+        if (fieldName != null) {
+            return fieldName;
+        }
+        if (isKeyContains) {
+            throw DiagnosticLog.error(DiagnosticErrorCode.DUPLICATE_FIELD, key);
+        }
+        return key;
+    }
+
+    public static long[] getSkipLinesFromStringConfigValue(String configValue) {
+        String[] parts = configValue.split(SKIP_LINE_RANGE_SEP);
+        if (parts.length != 2) {
+            throw DiagnosticLog.error(DiagnosticErrorCode.INVALID_RANGE_FOR_SKIPLINES);
+        }
+        try {
+            int start = Integer.parseInt(parts[0]);
+            int end = Integer.parseInt(parts[1]);
+            int size = end - start + 1;
+            if (size <= 0) {
+                throw DiagnosticLog.error(DiagnosticErrorCode.INVALID_RANGE_FOR_SKIPLINES);
+            }
+            long[] result = new long[size];
+            for (int i = 0; i < size; i++) {
+                result[i] = start + i;
+            }
+            return result;
+        } catch (NumberFormatException e) {
+            throw DiagnosticLog.error(DiagnosticErrorCode.INVALID_VALUE_FOR_SKIPLINES);
+        }
+    }
+
+    public static long[] getSkipDataRows(Object skipLines) {
+        long[] skipDataRows;
+        if (skipLines == null) {
+            return new long[]{};
+        }
+        if (skipLines instanceof BArray) {
+            BArray skipLinesArray = (BArray) skipLines;
+            if (skipLinesArray.getLength() == 0) {
+                return new long[]{};
+            }
+            skipDataRows = (skipLinesArray).getIntArray();
```

Comment:
Also may be we don't need this local variable assignment since we are going to return in the next line
```suggestion
            skipDataRows = skipLinesArray.getIntArray();
```